include/oneapi/dpl/internal/dynamic_selection_impl/policy_traits.h

```diff
@@ -0,0 +1,28 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_POLICY_TRAITS_H
+#define _ONEDPL_POLICY_TRAITS_H
+
+namespace oneapi {
+namespace dpl{
+namespace experimental{
+
+    template<typename Policy>
+    struct policy_traits{
+        using selection_type = typename std::decay<Policy>::type::selection_type;  //selection type
+        using resource_type = typename std::decay<Policy>::type::resource_type; //resource type
+
+        using wait_type = typename std::decay<Policy>::type::wait_type; //wait_type
```

Comment:
For `std::decay`, include `<type_traits>`. Also, use `std::decay_t<>` instead of `std::decay<>::type`

Reply:
Yeah, I also put the comment about `std::decay_t` somewhere else. Please apply to all files.

Reply:
Will do